ai_edge_torch/generative/examples/stable_diffusion/encoder.py

```diff
@@ -17,9 +17,60 @@
 from torch import nn
 from torch.nn import functional as F
 
-from ai_edge_torch.generative.examples.stable_diffusion.decoder import AttentionBlock  # NOQA
-from ai_edge_torch.generative.examples.stable_diffusion.decoder import ResidualBlock  # NOQA
-import ai_edge_torch.generative.utilities.loader as loading_utils
+from ai_edge_torch.generative.examples.stable_diffusion.attention import SelfAttention  # NOQA
+
+
+class AttentionBlock(nn.Module):
+
+  def __init__(self, channels):
+    super().__init__()
+    self.groupnorm = nn.GroupNorm(32, channels)
+    self.attention = SelfAttention(1, channels)
+
+  def forward(self, x):
+    residue = x
+    x = self.groupnorm(x)
+
+    n, c, h, w = x.shape
+    x = x.view((n, c, h * w))
+    x = x.transpose(-1, -2)
+    x = self.attention(x)
+    x = x.transpose(-1, -2)
+    x = x.view((n, c, h, w))
+
+    x += residue
+    return x
+
+
+class ResidualBlock(nn.Module):
+
+  def __init__(self, in_channels, out_channels):
```

Comment:
worth adding some comments here

Reply:
This update, moving from `decoder.py` to `encoder.py`, is because of refactoring `stable_diffusion/decoder.py`.
It is going to be deprecated soon. Maybe just leave it for now?

Reply:
yeah sounds good.